r/R/dplyr-functions.R

```diff
@@ -250,6 +250,27 @@ nse_funcs$is_logical <- function(x, n = NULL) {
   nse_funcs$is.logical(x)
 }
 
+# Create a data frame/tibble/struct column
+nse_funcs$tibble <- function(...) {
+  # use dots_list() because this is what tibble() uses to allow the
+  # useful shorthand of tibble(col1, col2) -> tibble(col1 = col1, col2 = col2_)
+  args <- rlang::dots_list(..., .named = TRUE)
+
+  build_expr(
+    "make_struct",
+    args = unname(args),
+    options = list(field_names = names(args))
+  )
+}
+
+nse_funcs$data.frame <- function(..., stringsAsFactors = FALSE) {
```

Comment:
What happens here is I supply any of the arguments that I could supply to `base::data.frame()`, (`check.rows`, `check.names`, or `fix.empty.names`)?

Reply:
I think so, yeah.  I know in the case of some functions we just leave the arguments out and it pushes it down the the R level, but I guess that only works when there is no `...` argument.